LION/optimizers/LIONsolver.py

```diff
@@ -141,35 +158,68 @@ def set_validation(
         """
         self.validation_loader = validation_loader
         self.validation_freq = validation_freq
-        self.validation_fn = validation_fn if validation_fn is not None else self.loss_fn
+        self.validation_fn = (
+            validation_fn if validation_fn is not None else self.loss_fn
+        )
         self.validation_fname = validation_fname
 
-    def set_testing(self, test_loader: DataLoader, testing_fn: Callable):
+    def set_testing(
+        self, test_loader: DataLoader, testing_fn: Optional[Callable] = None
+    ):
         """
         This function sets the testing data
         """
         self.test_loader = test_loader
-        self.testing_fn = testing_fn
+        self.testing_fn = testing_fn if testing_fn is not None else self.loss_fn
+
+    def set_saving(self, save_folder: str | pathlib.Path, final_result_fname: str):
+        if isinstance(save_folder, str):
+            save_folder = pathlib.Path(save_folder)
+        if not save_folder.is_dir():
+            raise ValueError(
+                f"Save folder '{save_folder}' is not a directory, failed to set saving."
+            )
+
+        self.save_folder = save_folder
+        self.final_result_fname = final_result_fname
+
+    def set_loading(self, load_folder: str | pathlib.Path, do_load: bool = False):
+        if isinstance(load_folder, str):
+            load_folder = pathlib.Path(load_folder)
+        if not load_folder.is_dir():
+            raise ValueError(
+                f"Save folder '{load_folder}' is not a directory, failed to set saving."
+            )
+
+        self.load_folder = load_folder
+        self.do_load_checkpoint = do_load
 
     def set_checkpointing(
         self,
         checkpoint_fname: str,
         checkpoint_freq: int = 10,
-        load_checkpoint: bool = False,
     ):
```

Comment:
new optional args:  save_path=None, load_path=None.

if checkpoint_fname is "full path" and save/load are None -> save_path/load_path are those
else read from this variables
else error.